### PR TITLE
Add rearest neighbor script

### PR DESCRIPTION
**Pull request by lavoiems**
Added a nearest-neighbor script using the same technique as in https://github.com/mrjel/noise-as-targets-tensorflow/blob/master/cifar10_natenc_getNeighbors.py

In the previous link, they use a l2 metric (line 63) to find the nearest-neighbor and they use a brute force algorithm (they compare a sample for every samples).

sklearn provide this utility and is optimized.

After some testing, this technique work fine but is not perfect since it only compare pixel wise values.

### CODE REVIEW

**Review by rdevon — APPROVED** [review]
Looks good
